File: 2.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        l=ListNode(0);
        l3=l;
        flag=0;
        
        while l1 or l2:
            node=ListNode(flag)
            if l1:
                node.val+=l1.val
                l1=l1.next
            if l2:
                node.val+=l2.val
                l2=l2.next
            flag=node.val/10
            node.val%=10
            l3.next=node
            l3=l3.next
        
        if flag:
            l3.next=ListNode(1)
        
        return l.next
                
            
            
        

File: 2.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode* l=new ListNode(0);
        ListNode* l3=l;
        int flag=0;
        while(l1!=NULL&&l2!=NULL){
            
            l3 ->next = new ListNode((l1->val+l2->val+flag)%10);
            flag=(l1->val+l2->val+flag)/10;
            l1=l1->next;
            l2=l2->next;
            l3=l3->next;
            
        }
        while(l1!=NULL){
            l3 ->next = new ListNode((l1->val+flag)%10);
            flag=(l1->val+flag)/10;
            l1=l1->next;
            l3=l3->next;
        }
        while(l2!=NULL){
            l3 ->next = new ListNode((l2->val+flag)%10);
            flag=(l2->val+flag)/10;
            l2=l2->next;
            l3=l3->next;   
        }
        if(flag==1){
            l3->next=new ListNode(1);
            l3=l3->next;
        }
        l3->next=NULL;
        
        return l->next;
    }
};
